traits/ctraits.c

```diff
@@ -4333,6 +4333,10 @@ post_setattr_trait_python(
     trait_object *trait, has_traits_object *obj, PyObject *name,
     PyObject *value)
 {
+    if (trait->py_post_setattr == NULL) {
```

Comment:
Is this code branch ever exercised? If it is, that's a coding error, so we should be raising some sort of exception here. But I suspect it's not.

What happens if you add an `assert(0)` here, compile with `-UNDEBUG`, and then run the test suite?

Reply:
This is not triggered if post_setattr is set to NULL whenever py_post_setattr is NULL. 
As you suspected, the branch execution was a coding error caused by changes in this PR.